lib/cell/concept.rb

```diff
@@ -5,8 +5,23 @@ class Concept < Cell::ViewModel
 
     # TODO: this should be in Helper or something. this should be the only entry point from controller/view.
     class << self
-      def class_from_cell_name(name)
-        name.camelize.constantize
+      def class_from_cell_name(name, controller)
+        @cell_class ||= begin
+          supports_path = controller.supports_path?
+          routes  = controller.respond_to?(:_routes)  && controller._routes
+          helpers = controller.respond_to?(:_helpers) && controller._helpers
+
+          Class.new(name.camelize.constantize) do
```

Comment:
Coping code form ActionView to Cells shows you misunderstanding of Cells and OOP in general.